internal/schemautil/userconfig/apiconvert/toapi.go

```diff
@@ -169,6 +169,15 @@ func itemToAPI(
 	// We omit the value if has no changes in the Terraform user configuration.
 	o := !d.HasChange(fks)
 
+	// We need to make sure that if there were any changes to the parent object, we also send the value, even if it
+	// was not changed.
+	//
+	// We check that there are more than three elements in the fk slice, because we don't want to send the value if
+	// the parent object is the root object.
+	if o && len(fk) > 3 && d.HasChange(fks[:strings.LastIndex(fks, ".0")]) {
+		o = false
+	}
```

Comment:
Please provide tests for this.

Reply:
https://github.com/aiven/terraform-provider-aiven/pull/1009#discussion_r1055746378